src/ray/protobuf/gcs_service.proto

```diff
@@ -530,6 +531,64 @@ autoscaler in a separate pod, if the user upgrades the ray version on the head
 pod autoscaler can crash (if the newer version of ray modified the messages
 below). */
 
+message InternalKVGetRequest {
+  bytes key = 1;
+}
+
+message InternalKVGetReply {
+  GcsStatus status = 1;
+  oneof optional_value {
```

Comment:
Why is this a `oneof`?

Reply:
In case you are getting non-existing value, it should be none. status here is for other usage like communication error I think?

Reply:
`GcsStatus` is actually for application-level `ray::Status` statuses, e.g. https://github.com/ray-project/ray/blob/9ceb0ca6410bb72e6c273ae8fe7f7af57d6649ba/src/ray/gcs/gcs_client/service_based_accessor.cc#L230-L234, and I think using that with a `ray::Status::NotFound()` status would be better than a pseudo-oneof in my opinion.

Reply:
I see. Thanks for the correction. Let me check how to update it. I agree status is a better choice. I was thinking that it's for other purposes.